docs/proposals/images/controlplane/controlplane-init-2.plantuml

```diff
@@ -0,0 +1,54 @@
+@startuml
+title User creates a Cluster with ControlPlane
+
+' -- GROUPS START ---
+
+box #lightgreen
+participant "API Server"
+end box
+
+box #violet
+participant "ControlPlane Controller"
+end box
+
+' -- GROUPS END ---
+
+note right of "ControlPlane Controller":Watches ControlPlane,\nCluster, and Machine events with mapFunc
+
+"API Server"-->>"ControlPlane Controller": Cluster Updated
+
+"ControlPlane Controller"-> "ControlPlane Controller":Enqueues ControlPlane Reconcile
+
+"ControlPlane Controller"-> "ControlPlane Controller":ControlPlane Controller Reconcile
+activate "ControlPlane Controller"
+
+note over "ControlPlane Controller": - ✅ ControlPlane.OwnerReferences \ncontains a Cluster
+
+"ControlPlane Controller"->"API Server": Get Cluster
+"ControlPlane Controller"<<--"API Server": Response
+
+note over "ControlPlane Controller": - ✅ Cluster.Status.InfraStructureReady is true\n- ✅ ControlPlane instance is valid
+
+"ControlPlane Controller"->"API Server": Get Machines maching label selector
+"ControlPlane Controller"<<--"API Server": Response
+
+opt ControlPlane.Spec.Replicas >= 1, no existing Machines found
+
+"ControlPlane Controller"->"API Server": Create KubeadmConfig from ControlPlane.Spec.KubeadmConfigSpec
+"ControlPlane Controller"<<--"API Server": Response
+
+"ControlPlane Controller"->"API Server": Create InfrastructureMachine from ControlPlane.Spec.InfrastructureTemplate
+"ControlPlane Controller"<<--"API Server": Response
+
+"ControlPlane Controller"->"API Server": Create Machine using refs from created KubeadmConfig and InfrastructureMachine resources
+"ControlPlane Controller"<<--"API Server": Response
```

Comment:
I assume this is done for each Machine, if so can we put this in a loop box?

Reply:
I specifically did the diagram around a single init to avoid confusion with introducing loops and branches. This is also why I called out that creating a control plane with replicas > 1 is the same as creating a control plane with a single replica followed by a scale operation.

Reply:
Got it, makes sense it wasn't clear though reading it through